api/internal/managers/app/install/util.go

```diff
@@ -32,10 +32,15 @@ func (m *appInstallManager) setupHelmClient() error {
 		return nil
 	}
 
+	k8sVersion, err := m.getK8sVersion()
```

Comment:
edge case: this logic can break later on upgrades if we start to deploy the app first before upgrading infra. for now it's fine

Reply:
why would we install the app before upgrading the infra? is this how it works today? i could see running the app preflights before installing the infra.

Reply:
Ah preflights is a good point. No that's not how it works today, we deploy infra first today